src/Settings/SettingsServiceProvider.php

```diff
@@ -28,5 +35,18 @@ public function register()
         });
 
         $this->app->alias(SettingsRepositoryInterface::class, 'flarum.settings');
+
+        $assets = function (Container $app) {
+            return $app->make(Factory::class)->disk('flarum-assets')->getDriver();
+        };
+
+        $this->app->when([
```

Comment:
I'm not 100% sure how I feel about having to enumerate every possible use of this interface here. Instead, what if we defined an AssetsFilesystemInterface, which extended FilesystemInterface, but had no body. Then, we could bind that to the $assets made above as a singleton, which should enable extensions to inject it, and gain access to assets?

Reply:
Yeah, I can relate. For now (and especially since this is about the `Application` contract right now) I can live with this. But we can (and should) revisit this when we tackle extensibility / scaling issues for the filesystem.

Reply:
Maybe this is actually a legit use-case for one of Illuminate's manager classes. /cc @luceos 

Probably not 100% the same use-cases, but at least we want to manage multiple "disks".

Reply:
I'm good with implementing this as a temporary solution, and refactoring later to allow file system driver extendability :+1:

Reply:
Agreeing with Sasha and yourself. Keep changes to a minimum and iterate to fulfill our future needs. This to me is okay.